syntax = "proto2";

package Example1;

message Company {
  optional int32 founded = 1;
  repeated int32 board_of_directors = 2;
  optional Financial financial = 11;
  repeated Employee all_employees = 12;
}

message Financial {
  repeated float quarterly_profits = 1;
  repeated float quarterly_revenues = 2;
}

message Employee {
  optional int32 id = 1;
  optional string name = 2;
  optional bool active = 3;
  repeated int32 active_direct_reports = 4;
  optional EnumX enumx = 5;
  repeated EnumY enumy = 6;
}

enum EnumX {
  ENUMX_UNKNOWN = 0;
  ENUMX_X1 = 1;
  ENUMX_X2 = 2;
}

enum EnumY {
  ENUMY_UNKNOWN = 0;
  ENUMY_Y1 = 1;
  ENUMY_Y2 = 2;
}
